bin/asinit

```diff
@@ -8,6 +8,30 @@ const colors = require("../cli/util/colors");
 const version = require("../package.json").version;
 const options = require("../cli/util/options");
 
+const commands = {
+  "npm": {
+    install: "npm install",
+    run: "npm run",
+    test: "npm test"
+  },
+  "yarn": {
+    install: "yarn install",
+    run: "yarn",
+    test: "yarn test"
+  },
+  "pnpm": {
+    install: "pnpm install",
+    run: "pnpm run",
+    test: "pnpm test"
+  }
+};
+
+let pm = "npm";
+if (process.env.npm_config_user_agent.test(/\byarn\//))
+  pm = "yarn";
+else if (process.env.npm_config_user_agent.test(/\bpnpm\//))
+  pm = "pnpm";
```

Comment:
Did a quick web search for `npm_config_user_agent` and found [one issue](https://github.com/lerna/lerna/issues/1369) describing that it may be `undefined` in some uncommon scenarios. Not sure how relevant that is still, but perhaps it's good to add a sanity check?

```suggestion
if (typeof process.env.npm_config_user_agent === "string") {
  if (process.env.npm_config_user_agent.test(/\byarn\//)) {
    pm = "yarn";
  } else if (process.env.npm_config_user_agent.test(/\bpnpm\//)) {
    pm = "pnpm";
  }
}
```

Reply:
or:
```js
const ua = process.env.npm_config_user_agent || "";

let pm = "npm";
if (ua.test(/\byarn\//)) {
  pm = "yarn";
} else if (ua.test(/\bpnpm\//)) {
  pm = "pnpm";
}
```

Reply:
Good idea 👍